app/models/salesforce/case.rb

```diff
@@ -83,5 +76,18 @@ def self.contact_id_for(user)
         user.create_salesforce_contact
       end
     end
+
+    private
+
+    def payload
+      default = {
+        ContactId: self.contact_id,
+        Subject: self.subject,
+        Description: self.description,
+        Origin: SiteSetting.salesforce_case_origin,
+      }
+
+      DiscoursePluginRegistry.apply_modifier(:salesforce_case_payload, default)
```

Comment:
Would it be useful to pass in other information to the modifier like the `topic` or `current_user` ?

Reply:
`topic` might be useful since the case is based on a topic. `current_user`? not so sure. I'll add `topic` and hold on till there is an actual use case for `current_user`.